packages/next-remoter/src/composable/useRouteBasedTools.ts

```
@@ -76,11 +128,11 @@ export function useRouteBasedTools(options: {
     })
 
     // 2. 更新 ignoreToolnames：仅针对路由绑定工具做过滤
+    // 注意：虽然 CustomAgentModelProvider 会在 prepareStep 中覆盖 activeTools，
+    // 但保留此处的 ignoreToolnames 同步可以确保在基础场景下也有正确的工具屏蔽兜底。
     const ignore = new Set<string>(agent.ignoreToolnames as string[])
     // 将所有路由工具先加入忽略集合，然后移除当前激活的
-    routeMap.forEach((_route, toolName) => {
-      ignore.add(toolName)
-    })
+    routeToolIds.forEach((id) => ignore.add(id))
```

Biome (2.4.6)

[error] This callback passed to forEach() iterable method should not return a value.

(lint/suspicious/useIterableCallbackReturn)